# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerDocumentListener.cpp
# Commit: 7802e5f1bd13
# Full Hash: 7802e5f1bd1395f19f2d78636787c41f64192bd7
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-04-19 15:22:44
# Regressor Bug: 1769913
# File Overlap Count: 3
# Description:
#   Bug 1769913 - P3 Remove WorkerRunnable::mWorkerPrivate. r=dom-worker-reviewers,asuth
#   
#   WorkerRunnable no longer keeps a raw pointer(mWorkerPrivate) for the associated WorkerPrivate in this patch.
#   Removing the WorkerRunnable::mWorkerPrivate needs to fix the following problems.
#   
# ==============================================================================

diff -r ffc69aa8f2c3 -r 7802e5f1bd13 dom/workers/WorkerDocumentListener.cpp
--- a/dom/workers/WorkerDocumentListener.cpp	Fri Apr 19 09:41:57 2024 +0000
+++ b/dom/workers/WorkerDocumentListener.cpp	Fri Apr 19 09:41:58 2024 +0000
@@ -55,8 +55,7 @@
   class VisibleRunnable final : public WorkerThreadRunnable {
    public:
     VisibleRunnable(WorkerPrivate* aWorkerPrivate, bool aVisible)
-        : WorkerThreadRunnable(aWorkerPrivate, "VisibleRunnable"),
-          mVisible(aVisible) {}
+        : WorkerThreadRunnable("VisibleRunnable"), mVisible(aVisible) {}
 
     bool WorkerRun(JSContext* aCx, WorkerPrivate* aWorkerPrivate) {
       WorkerGlobalScope* scope = aWorkerPrivate->GlobalScope();
@@ -70,7 +69,7 @@
   };
 
   auto runnable = MakeRefPtr<VisibleRunnable>(mWorkerRef->Private(), aVisible);
-  runnable->Dispatch();
+  runnable->Dispatch(mWorkerRef->Private());
 }
 
 void WorkerDocumentListener::SetListening(uint64_t aWindowID, bool aListen) {